tsconfig.json

```diff
@@ -0,0 +1,9 @@
+{
+    "compilerOptions": {
+      "esModuleInterop": true,
+      "lib": ["es2017"],
+      "module": "commonjs",
+      "resolveJsonModule": true,
+      "target": "es2017"
```

Comment:
Reading between the lines, `es2017` is good enough for node 8, but then we still need babel to make it browser-compatible?

It might be worth adding a minimal eslint config for `dist` just to make sure we have valid es5 coming out.

Reply:
Was thinking we will have another tsconfig for the browser build. This base one is for ts-node/register to use for unit tests / feature tests